addons/html_editor/static/src/local_overlay_container.js

```diff
@@ -0,0 +1,26 @@
+import { Component } from "@odoo/owl";
+import { MainComponentsContainer } from "@web/core/main_components_container";
+import { useForwardRefToParent } from "@web/core/utils/hooks";
+import { registry } from "@web/core/registry";
+import { useRegistry } from "@web/core/registry_hook";
+
+export class LocalOverlayContainer extends MainComponentsContainer {
+    static template = "html_editor.LocalOverlayContainer";
+    static props = {
+        localOverlay: { type: Function, optional: true },
+        identifier: { type: String, optional: true },
+    };
+    static defaultProps = {
+        identifier: "overlay_components",
+    };
+
+    setup() {
+        const overlayComponents = registry.category(this.props.identifier);
```

Comment:
Use a reactive props that has the Components instead of using a registry. (you can add a todo if you don't have time to refactor)